framework/aster-frame/src/mm/io.rs

```diff
@@ -341,64 +603,130 @@ impl<'a> VmWriter<'a> {
         self
     }
 
-    /// Writes data from the reader until one of the two conditions is met:
+    /// Writes data from the reader into this writer until one of the three conditions is met:
     /// 1. The writer has no available space.
     /// 2. The reader has no remaining data.
+    /// 3. The writing encounters a page fault that cannot be handled.
     ///
     /// Returns the number of bytes written.
     ///
     /// It pulls the number of bytes data from the reader and
     /// fills in the writer with the number of bytes.
-    pub fn write(&mut self, reader: &mut VmReader<'_>) -> usize {
+    ///
+    /// The writing instruction is forbidden if the targets of reader and writer
+    /// are both user space.
+    pub fn write<R: 'static>(&mut self, reader: &mut VmReader<'_, R>) -> usize {
```

Comment:
Seems that `writer.write(reader)` is effectively the same as `reader.read(writer)`.

Why not:

```rust
    pub fn write<R: 'static>(&mut self, reader: &mut VmReader<'_, R>) -> usize {
        reader.read(self)
    }
```

, saving 50% code size.

It is even preferable only providing `writer.write(reader)`.